ruby-challenges.rb

```diff
@@ -1,32 +1,91 @@
 # ASSESSMENT 5: Ruby Coding Practical Questions
 # MINASWAN
 
-# --------------------1) Create a method that takes in an array of words and a single letter and returns an array of all the words containing that particular letter. Use the test variables provided.
+# --------------------1)
+def filter_by_letter(words, letter)
+    filtered_words = []
 
-beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
+    words.each do |word|
+       if word.include?(letter)
+        filtered_words << word
+       end
+    end
+    filtered_words
+end
 
-filter_letter_o = 'o'
-# Expected output: ['coffee', 'soda water']
-filter_letter_t = 't'
-# Expected output: ['tea', 'water', 'soda water']
+describe 'filter_by_letter' do
+    beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
+    
+    it 'returns words that have the letter "o"' do
+        filter_letter_o = 'o'
+        result = filter_by_letter(beverages_array, filter_letter_o)
+        expect(result).to eq(['coffee','soda water'])
+    end
 
+    it 'returns words that have the letter "t"' do
+        filter_letter_t = 't'
+        result = filter_by_letter(beverages_array, filter_letter_t)
+        expect(result).to eq(['tea', 'water', 'soda water'])
+    end
 
-# -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
-# HINT: Google 'ruby get rid of nested arrays'
+end
 
-us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
-# Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
 
+# -------------------2)
+def states_in_order(hash)
+    values = hash.values.flatten
```

Comment:
You can chain .sort at the end of this as well, and then remove all the variables you made